editors/emacs/prettier-js.el

```diff
@@ -219,4 +219,13 @@ function."
      (delete-file bufferfile)
      (delete-file outputfile)))
 
+;;;###autoload
+(define-minor-mode prettier-mode
+  "Runs prettier on file save when this mode is turned on"
+  :lighter " prettier"
+  :global t
+  (if prettier-mode
+      (add-hook 'before-save-hook 'prettier-before-save)
```

Comment:
I find a bit confusing the fact that `prettier-mode` depends on the `prettier-target-mode` variable.

Now that we have a minor mode, we can just hook it to any mode we want instead of limiting it to just one mode. This can be useful if we are using multiple modes for javascript, ie: js2-mode for javascript files and web-mode for jsx files.

We can fix this by hooking here the `prettier` function instead of `prettier-before-save`.

What do you think?